presto-docs/src/main/sphinx/admin/verifier.rst

```diff
@@ -0,0 +1,257 @@
+    * See `Failure Resolution`_ for automatic resolving query failures.
+
+* **Results Comparison**
+    * For ``Select``, ``Insert``, and ``CreateTableAsSelect`` queries, results are written into temporary tables.
+    * Construct and run the checksum queries for both control and test.
+    * Verify table schema and row count are the same for the control and the test result table.
+    * Verify checksums are matching for each column. See `Column Checksums`_ for special handling of different column types.
+    * See `Determinism`_ for handling of non-deterministic queries.
+
+* **Emitting Results**
+    * Verification results can be exported as ``JSON``, or human readable text.
+
+Column Checksums
+----------------
+For each column in the control/test query, one or more columns maybe generated in the checksum
+queries.
+
+* **Floating Point Columns**
+    * For ``DOUBLE`` and ``REAL`` columns, 4 columns are generated for verification:
+       * Sum of the finite values of the column
+       * ``NAN`` count of the column
+       * Positive infinity count of the column
+       * Negative infinity count of the column
+    * Check if ``NAN`` count, positive and negative infinity count matches.
+    * Check the nullity of control sum and test sum.
+    * If either control mean or test mean very close 0, check if both are close to 0.
+    * Check the relative error between control sum and test sum.
+* **Array Columns**
+    * 2 columns are generated for verification:
+       * Sum of the cardinality
+       * Array checksum
+    * For an array column ``arr`` of type ``array(E)``:
+       * If ``E`` is not orderable, array checksum is ``checksum(arr)``.
+       * If ``E`` is orderable, array checksum ``coalesce(checksum(try(array_sort(arr))), checksum(arr))``.
+* **Map Columns**
+    * 4 columns are generated for verification:
+       * Sum of the cardinality
+       * Checksum of the map
+       * Array checksum of the kep set
+       * Array checksum of the value set
+* **Row Columns**
+    * Checksum row fields recursively, as according to the type of the fields.
+* For all other column types, generate a simple checksum using the :func:`checksum` function.
+
+Determinism
+-----------
+A result mismatch, either a row count mismatch or a column mismatch, can be caused by
+non-deterministic query features. To avoid false alerts, we perform determinism analysis
+for the control query. If a query is found non-deterministic, we skip the verification as it
+does not provide insights.
+
+Determinism analysis follows the following steps. If a query is found non-deterministic at any
+point, the analysis will conclude.
+
+* A list of non-deterministic catalogs can be specified. If a query references any table from a
+  non-deterministic catalog, the query is considered non-deterministic.
+* Run the control query again, and the compare the results with the initial control query run.
```

Comment:
typo: "Run the control query again, and the compare" -> "Run the control query again and compare"